examples/example1.cpp

```diff
@@ -110,10 +96,12 @@ int main(int argc, char **argv)
   node6.add_child(&node8);
 
   auto api = get_c_api();
-  auto results = api.find(&root, "/compilation_unit//identifier");
-  for (auto node : results)
-  {
-    std::cout << ((Node *)node)->internal_type << std::endl;
+  void *results[100];
```

Comment:
* This should be `Node *results[100]` and casted to `(void *)` when calling `node_find`, not the other way around.
* The result space and limits should not be handled by the library user. Having to guess the number of results beforehand is error-prone and a waste of memory. `node_find` may need to return another data structure you can use to iterate throgh the results.